Add tests for `ListBucketClass`

Ashughorla opened the pull request:
# Proposed Changes

- Add tests for list bucket class

Fixes #363

afritzler reviewed (changes requested):
1. Can you please remove the `controller-gen` in your projects `bin` folder and re-run `make generate`.
2. Can you please run `go mod tidy` to cleanup the `go.sum` diffs.
3. Is there a reason why we moved the BucketClass creation into each test? I would have kept it in the `suite_test.go`.

Ashughorla commented:
> 1. Can you please remove the `controller-gen` in your projects `bin` folder and re-run `make generate`.
> 2. Can you please run `go mod tidy` to cleanup the `go.sum` diffs.
> 3. Is there a reason why we moved the BucketClass creation into each test? I would have kept it in the `suite_test.go`.

For point
1. Addressed.
2. Done `go mod tidy` and `make test`. It doesn't show any `go.sum` diffs.
3. To test scenario to check empty BucketClass list.
